src/internal.c

```diff
@@ -22454,6 +22499,16 @@ int SendData(WOLFSSL* ssl, const void* data, int sz)
         ssl->error = ret;
         return WOLFSSL_FATAL_ERROR;
     }
+#if defined(WOLFSSL_TLS13) && defined(WOLFSSL_TLS13_AUTO_REKEY)
+    if (IsAtLeastTLSv1_3(ssl->version) &&
+        (ssl->options.rekey != REKEY_IN_PROCESS) && Tls13_NeedRekey(ssl,
+            ssl->keys.maxInvocations, ssl->keys.sequence_number_hi,
+            ssl->keys.sequence_number_lo, 0)) {
+        ret = wolfSSL_update_enc_keys(ssl);
+        if (ret != WOLFSSL_SUCCESS)
+            return ret;
+    }
+#endif
```

Comment:
DTLS 1.3 uses `ssl->keys.curSeq` for the peer's current sequence number. Please check that this works for DTLS 1.3 too.